StRoot/StFcsWaveformFitMaker/PeakAna.h

```diff
@@ -0,0 +1,351 @@
+
+@[January 29, 2022]
+> Changes to how drawing functions work and a compare for PeakWindow
+
+@[January 28, 2022]
+> Fixes to peak tunnel method includng adding a tunnel sigma variable since using baseline sigma did not work as expected
+
+@[January 24, 2022]
+> Added draw methods
+
+@[January 23, 2022]
+> Added the peak tunneling method for skipping over noise data
+
+@[January 19, 2022]
+> First instance of this class that is adopted from work on *StFcsPulseFit*
+
+The purpose of this class is to find peaks in a sample data and the range of those peaks; i.e. where the first derivative is zero and second derivate changes sign from positive to negative and back to positive as this indicates a local max. 
+
+There is a helper data class *PeakWindow* which holds the data and does not inherit from TObject
+
+@[January 21, 2022]>Inherit PeakAna from TGraph??
+
+
+*/
+
+#ifndef STROOT_STFCSWAVEFORMFITMAKER_PEAKANA_H_
+#define STROOT_STFCSWAVEFORMFITMAKER_PEAKANA_H_
+
+//C++ Headers
+#include <iostream>
+#include <vector>
+#include <utility> //std::move (Needs C++ 11) and std::pair
+#include <iterator>
+
+//ROOT Headers
+#include "TObject.h"
+#include "TAttLine.h"
+#include "TKey.h"
+#include "TH1.h"
+#include "TPaveText.h"
+
+//Custom Headers
+#include "PeakWindow.h"
+#include "PeakAnaVirtualPainter.h"
+
+class PeakAna : public TObject
+{
+public:
+  PeakAna();
+  PeakAna( int size, double* xvals, double* yvals );//Array of x and y values with known size
+  PeakAna( TGraph* Sig );//Construct with a known TGraph
+  PeakAna( TH1* hist);
+  PeakAna(const PeakAna &OldAna,TGraph* graph=0); //Copy Constructor
+  PeakAna& operator=(const PeakAna& rhs);//Assignment operator
+  virtual ~PeakAna();
+  
+  virtual void Copy(TObject& obj) const;//Clone graph as opposed to just copying the pointer
+  virtual TObject* Clone(const char* newname) const;
+  
+  virtual void AddPeakStats(TPaveText* pave, const char* opt="");
+  //Peak finding methods are independent of the class
+  static TGraph* ConvertHistToGraph(TH1* graph, UInt_t numavgs=1);//The second argument can be used to do a running average over 'numavgs' number of bins (1 means no average)
+  static TGraph* ConvertPeaksToGraph(const std::vector<PeakWindow> &Peaks);//The purpose of this function is to help with really noisy data where the "peaks" are really just noise. Essentially the idea is that each "peak" should actually be a point on a kind of "running average" in this way you capture the overall shape of the data without noise
+  void ConvertPeaksToGraph();//same as above but creates new graph that replaces old one
+  static PeakAna* ConvertPeaksToAna(const PeakAna &Ana);//same as above but returns new PeakAna with same settings without modifying old one
+  PeakAna* ConvertPeaksToAna();//same as above but returns new PeakAna with same settings without modifying current one
```

Comment:
Doxygen style comments please

Reply:
I am still learning the syntax for doxygen style comments. Is there a way to generate the html for a given file or files so that I can see how it is working?